extensions/wikia/Email/EmailController.class.php

```diff
@@ -481,17 +481,34 @@ protected function getAvatarURL( \User $user ) {
 	}
 
 	protected function findUserFromRequest( $paramName, \User $default = null ) {
-		$userName = $this->getRequest()->getVal( $paramName );
-		if ( empty( $userName ) ) {
+		$user = $this->getRequest()->getVal( $paramName );
+		if ( empty( $user ) ) {
 			return $default;
 		}
 
 		// Allow an anonymous user to be specified
-		if ( $userName == self::ANONYMOUS_USER_ID ) {
+		if ( $user == self::ANONYMOUS_USER_ID ) {
 			return \User::newFromId( 0 );
 		}
 
-		return $this->getUserFromName( $userName );
+		if ( is_numeric( $user ) ) {
```

Comment:
This wont work correctly all the time. There are users with numeric username.
For instance,
```sql
mysql@slave.db-sharedb.service.consul[wikicities]>select user_name from user where user_name = '12';
+-----------+
| user_name |
+-----------+
| 12        |
+-----------+
1 row in set (0.01 sec)
```

How about limiting the input to only username?